sonar-duplications/src/main/java/org/sonar/duplications/index/PackedMemoryCloneIndex.java

```diff
@@ -104,31 +109,102 @@ public PackedMemoryCloneIndex(int hashBytes, int initialCapacity) {
     List<Block> result = new ArrayList<>();
     int realIndex = resourceIdsIndex[index];
     while (index < size && FastStringComparator.INSTANCE.compare(resourceIds[realIndex], resourceId) == 0) {
-      // extract block (note that there is no need to extract resourceId)
-      int offset = realIndex * blockInts;
+      result.add(getBlock(realIndex, resourceId));
+
+      index++;
+      realIndex = resourceIdsIndex[index];
+    }
+    return result;
+  }
+
+  private Block createBlock(int index, String resourceId, @Nullable ByteArray byteHash) {
+    int offset = index * blockInts;
+
+    if (byteHash == null) {
       int[] hash = new int[hashInts];
       for (int j = 0; j < hashInts; j++) {
         hash[j] = blockData[offset++];
       }
-      int indexInFile = blockData[offset++];
-      int firstLineNumber = blockData[offset++];
-      int lastLineNumber = blockData[offset++];
-      int startUnit = blockData[offset++];
-      int endUnit = blockData[offset];
-
-      Block block = blockBuilder
-        .setResourceId(resourceId)
-        .setBlockHash(new ByteArray(hash))
-        .setIndexInFile(indexInFile)
-        .setLines(firstLineNumber, lastLineNumber)
-        .setUnit(startUnit, endUnit)
-        .build();
-      result.add(block);
+      byteHash = new ByteArray(hash);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "byteHash". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1226)